src/hb-serialize.hh

```diff
@@ -41,6 +41,14 @@
  * Serialize
  */
 
+enum hb_serialize_error_type_t {
+  HB_SERIALIZE_ERR_NONE =            0x00000000u,
+  HB_SERIALIZE_ERR_OTHER =           0x00000001u,
+  HB_SERIALIZE_ERR_OFFSET_OVERFLOW = 0x00000002u,
+  HB_SERIALIZE_ERR_OUT_OF_ROOM =     0x00000004u,
+};
+HB_MARK_AS_FLAG_T (hb_serialize_error_type_t);
```

Comment:
The enum name and the value names should match. Rename enum to `hb_serialize_error_t` and values to `HB_SERIALIZE_ERROR_...`.

Now, since this is internal and won't be exposed to C API, you can use enum classes with shorter value names, if that feels better. Guess not here, since the enum name needs to be repeated all the time anyway.